src/drivers/intel/cavs/idc.c

```diff
@@ -217,6 +217,107 @@ static void idc_ipc(void)
 	ipc_cmd(hdr);
 }
 
+/**
+ * \brief Executes IDC component params message.
+ * \param[in] comp_id Component id to have params set.
+ * \return Error code.
+ */
+static int idc_params(uint32_t comp_id)
+{
+	struct ipc *ipc = ipc_get();
+	struct ipc_comp_dev *ipc_dev;
+	struct idc *idc = *idc_get();
+	struct idc_payload *payload = idc->payload + cpu_get_id();
```

Comment:
Can we have func/macro for getting payload for given core?

Reply:
Done.